src/3d/qgs3dmapsettings.h

```diff
@@ -436,27 +436,44 @@ class _3D_EXPORT Qgs3DMapSettings : public QObject, public QgsTemporalRangeObjec
 
     /**
      * Returns list of point lights defined in the scene
-     * \since QGIS 3.6
+     * \deprecated use lightSources() instead
      */
-    QList<QgsPointLightSettings> pointLights() const { return mPointLights; }
+    Q_DECL_DEPRECATED QList<QgsPointLightSettings> pointLights() const SIP_DEPRECATED;
 
     /**
      * Returns list of directional lights defined in the scene
-     * \since QGIS 3.16
+     * \deprecated use lightSources() instead
      */
-    QList<QgsDirectionalLightSettings> directionalLights() const { return mDirectionalLights; }
+    Q_DECL_DEPRECATED QList<QgsDirectionalLightSettings> directionalLights() const SIP_DEPRECATED;
 
     /**
      * Sets list of point lights defined in the scene
-     * \since QGIS 3.6
+     * \deprecated use setLightSources() instead
      */
-    void setPointLights( const QList<QgsPointLightSettings> &pointLights );
+    Q_DECL_DEPRECATED void setPointLights( const QList<QgsPointLightSettings> &pointLights ) SIP_DEPRECATED;
 
     /**
      * Sets list of directional lights defined in the scene
-     * \since QGIS 3.16
+     * \deprecated use setLightSources() instead
+     */
+    Q_DECL_DEPRECATED void setDirectionalLights( const QList<QgsDirectionalLightSettings> &directionalLights ) SIP_DEPRECATED;
```

Comment:
I would just remove these methods rather just deprecating them - the whole 3D api is still considered unstable...

Reply:
Done